k8s_pod_watcher.go

```diff
@@ -177,20 +214,36 @@ func (p *PodWatcher) resync(ctx context.Context, cbChans []chan<- PodEvent) (str
 		return "", fmt.Errorf("failed pod listing: %w", initListerr)
 	}
 	podNames := make([]string, len(initPods.Items))
+
+	synthPodEvents := make([]PodEvent, 0, len(initPods.Items))
 	for i, pod := range initPods.Items {
 		podNames[i] = pod.Name
-		if ev := p.tracker.synthesizeEvent(&pod); ev != nil {
-			for _, ch := range cbChans {
-				ch <- ev
-			}
+		if ev := p.tracker.synthesizeEvent(&initPods.Items[i]); ev != nil {
+			synthPodEvents = append(synthPodEvents, ev)
 		}
 	}
+
+	// We need to know how many pods died in order to set continues properly on the non-delete
+	// events, but we can't update the tracker state until after we've synthesized all the other
+	// events.
 	deadPods := p.tracker.findRemoveDeadPods(podNames)
+
+	for i, ev := range synthPodEvents {
+		continues := len(deadPods) > 0 || i < len(synthPodEvents)-1
```

Comment:
Let's add a comment here about the last event setting continues to true.

Reply:
done.